jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1845,10 +1847,52 @@ public static URI toURI(String resource)
     {
         Objects.requireNonNull(resource);
 
-        // Only try URI for string for known schemes, otherwise assume it is a Path
-        return (ResourceFactory.isSupported(resource))
-            ? correctFileURI(URI.create(resource))
-            : Paths.get(resource).toUri();
+        if (URIUtil.hasScheme(resource))
+        {
+            try
+            {
+                URI uri = new URI(resource);
+
+                if (ResourceFactory.isSupported(uri))
+                    return correctFileURI(uri);
+
+                // We don't have a supported URI scheme
+                if (uri.getScheme().length() == 1)
+                {
+                    // Input is a possible Windows path disguised as a URI "D:/path/to/resource.txt".
+                    try
+                    {
+                        return toURI(Paths.get(resource).toUri().toASCIIString());
+                    }
+                    catch (InvalidPathException ignore)
+                    {
+                        // ignore
+                    }
+                }
+
+                // If we reached this point, that means the input String has a scheme,
+                // and is not recognized as supported by the registered schemes in ResourceFactory.
+                throw new IllegalArgumentException("URI scheme not supported: " + resource);
```

Comment:
Tell me again why we can't just `return uri` here?     We have done our job of intercepting any `Path`s (windows or otherwise) and we have a URI that has been successfully parsed from the string, so why not return it?

If there is a good reason why not, then it needs to be in the comment here.  I.e it has a scheme that is not recognized ..... and the problem with that is ???

Reply:
If the URI scheme isn't recognized then it cannot be used by the Resource system.
If a user wants to use a new scheme, lets say `foo://1234`, then it's up to them to decide if they want to support that via the java FileSystem API or the URL API.
That is done via the call to `ResourceFactory.registerResourceFactory(String scheme, ResourceFactory resourceFactory)` where they can specify how they want it supported.

For FileSystem based, they can use `ResourceFactory.registerResourceFactory("foo", new PathResourceFactory());` (like graalvm does)
For URL based, they can use `ResourceFactory.registerResourceFactory("foo", new URLResourceFactory());` (like osgi does)

If the scheme isn't registered, it's useless, and nothing in the Resource layer can use it. (until it is registered).

Reply:
see the `ResourceFactoryTest` for the tests `testCustomUriSchemeRegistered` and `testRegisterHttpsUrlFactory`